includes/Modules/Analytics.php

```diff
@@ -767,6 +775,86 @@ function ( $order_def ) {
+				}
+				if ( ! isset( $data['timezone'] ) ) {
+					/* translators: %s: Missing parameter name */
+					return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'timezone' ), array( 'status' => 400 ) );
+				}
+				$credentials = new Credentials( new Encrypted_Options( $this->options ) );
+				if ( ! $credentials->has() ) {
+					return new WP_Error( 'missing_credentials', __( 'Missing credentials..', 'google-site-kit' ), array( 'status' => 400 ) );
+				}
+
+				$account = new Google_Service_Analytics_Account();
+				$account->setName( $data['accountName'] );
+
+				$property = new Google_Service_Analytics_Webproperty();
+				$property->setName( $data['propertyName'] );
+				$property->setWebsiteUrl( $this->context->get_reference_site_url() );
+
+				$profile = new Google_Service_Analytics_Profile();
+				$profile->setName( $data['profileName'] );
+				$profile->setTimezone( $data['timezone'] );
+
+				$account_ticket = new AccountTicket();
+				$account_ticket->setAccount( $account );
+				$account_ticket->setWebproperty( $property );
+				$account_ticket->setProfile( $profile );
+				$account_ticket->setRedirectUri( $this->get_provisioning_redirect_uri() );
+
+				// Add site id and secret.
+				$creds = $credentials->get();
+				$account_ticket->setSiteId( $creds['oauth2_client_id'] );
+				$account_ticket->setSiteSecret( $creds['oauth2_client_secret'] );
+
+				$restore_defer = $this->with_client_defer( false );
+				try {
+					$analytics_service = $this->get_service( 'analyticsprovisioning' );
+
+					$custom_provisioning = new Provisioning(
+						$analytics_service,
+						$analytics_service->serviceName, // phpcs:ignore WordPress.NamingConventions.ValidVariableName
+						'provisioning',
+						array(
+							'methods' => array(
+								'createAccountTicket' => array(
+									'path'       => 'provisioning/createAccountTicket',
+									'httpMethod' => 'POST',
+									'parameters' => array(),
+								),
+							),
+						)
+					);
+
+					// Use a custom provisioning method that accepts site id and secret.
+					$analytics_service->provisioning = $custom_provisioning;
+
+					$account_ticket = $analytics_service->provisioning->createAccountTicket( $account_ticket );
+				} catch ( Exception $e ) {
+					$restore_defer();
+					return $this->exception_to_error( $e, $data->datapoint );
+				}
+				$restore_defer();
+				$account_ticket_id = $account_ticket->getId();
+				if ( empty( $account_ticket_id ) ) {
+					return new WP_Error( 'analytics-provisioning', 'no-account-ticket-id', array( 'status' => 500 ) );
+				}
+
+				// Cache the create ticket id long enough to verify it upon completion of the terms of service.
+				set_transient( $self::ANALYTICS_PROVISIONING_ACCOUNT_TICKET_ID . '::' . get_current_user_id(), $account_ticket_id, 15 * MINUTE_IN_SECONDS );
+				return $account_ticket;
```

Comment:
The function in this datapoint needs to be split (we have `create_data_request` and `parse_data_response`).

* `create_data_request` should handle everything up to the request, it needs to `return $analytics_service->provisioning->createAccountTicket( $account_ticket )`. No try-catch and no `$restore_defer` should be in that method, it's all handled by the general datapoint wrapper.
* `parse_data_response` should further handle the `$response`, which will be that account ticket object. So the part where it sets the transient and then just passes through the account ticket should be in the response handler.

See other datapoints that work like that. Essentially every datapoint that as its main purpose makes an API request should be split like that, if it also needs to do something specific with the response. The main purpose of this is to support batching, which we may phase out at some point, but it's still there generally, so let's follow our API here.

Reply:
Ah right, thanks for clarifying... changing this.

Reply:
Ah, right. Removed this bit and calling/returning at once.